pkg/clients/gcp/subnetwork/subnetwork_test.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package subnetwork
+
+import (
+	"sort"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"google.golang.org/api/compute/v1"
+
+	"github.com/crossplaneio/crossplane/gcp/apis/compute/v1alpha1"
+)
+
+func TestGCPSubnetworkSpec_GenerateSubnetwork(t *testing.T) {
+	cases := map[string]struct {
+		in  v1alpha1.GCPSubnetworkSpec
+		out *compute.Subnetwork
+	}{
+		"FilledGeneration": {
+			in: v1alpha1.GCPSubnetworkSpec{
+				Description:           "some desc",
+				Name:                  "some-name",
+				EnableFlowLogs:        true,
+				IPCidrRange:           "10.0.0.0/9",
+				Network:               "test-network",
+				PrivateIPGoogleAccess: true,
+				Region:                "test-region",
+				SecondaryIPRanges: []*v1alpha1.GCPSubnetworkSecondaryRange{
+					{
+						RangeName:   "zzaa",
+						IPCidrRange: "10.0.0.0/9",
+					},
+					{
+						RangeName:   "aazz",
+						IPCidrRange: "10.0.0.1/9",
+					},
+				},
+			},
+			out: &compute.Subnetwork{
+				Description:           "some desc",
+				Name:                  "some-name",
+				EnableFlowLogs:        true,
+				IpCidrRange:           "10.0.0.0/9",
+				Network:               "test-network",
+				PrivateIpGoogleAccess: true,
+				Region:                "test-region",
+				SecondaryIpRanges: []*compute.SubnetworkSecondaryRange{
+					{
+						RangeName:   "aazz",
+						IpCidrRange: "10.0.0.1/9",
+					},
+					{
+						RangeName:   "zzaa",
+						IpCidrRange: "10.0.0.0/9",
+					},
+				},
+			},
+		},
+	}
+
+	for name, tc := range cases {
+		t.Run(name, func(t *testing.T) {
+			r := GenerateSubnetwork(tc.in)
+			sort.SliceStable(r.SecondaryIpRanges, func(i, j int) bool {
+				return r.SecondaryIpRanges[i].RangeName > r.SecondaryIpRanges[j].RangeName
+			})
+			sort.SliceStable(tc.out.SecondaryIpRanges, func(i, j int) bool {
+				return tc.out.SecondaryIpRanges[i].RangeName > tc.out.SecondaryIpRanges[j].RangeName
+			})
```

Comment:
I have changed it here in the test to use cmp's option, but I think it's better to keep using standard library `sort` in the productive code (`pkg/clients/gcp/subnetwork/subnetwork.go`) in order not to force people have cmp as dependency when they import our package, just because of this.

Reply:
Thanks! Agreed that it's better to use regular sort in the actual non-test code.